providers/cloudflare/rest.go

```diff
@@ -150,9 +150,12 @@ func (c *cloudflareProvider) createRec(rec *models.RecordConfig, domainID string
 			} else if rec.Type == "DS" {
 				cf.Data = cfDSData(rec)
 			}
-			resp, err := c.cfClient.CreateDNSRecord(context.Background(), domainID, cf)
-			id = resp.Result.ID
-			return err
+			if resp, err := c.cfClient.CreateDNSRecord(context.Background(), domainID, cf); err != nil {
+				return err
+			} else {
+				id = resp.Result.ID
```

Comment:
I'm probably missing something obvious but why set a variable just before a "return nil" statement?  I suspect either the `id =` or `return nil` isn't needed.

Reply:
As far as i see, `id =` is required at L164 to enable the proxy mode. And `return nil` is required as the function for `F` requires a return type:

```go
arr := []*models.Correction{{
	Msg: fmt.Sprintf("CREATE record: %s %s %d%s %s", rec.GetLabel(), rec.Type, rec.TTL, prio, content),
	F: func() error {}
})
```

Reply:
OH!  The "id" variable is stored outside of the function!  I get it.  It's being updated by side-effect.

Could you add a comment like, "Updating id (from the outer scope) by side-effect" for future me ("future me" has terrible memory).

Thanks!